org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/internal/workspace/SynchronizingRenamedProject.groovy

```diff
@@ -47,7 +49,7 @@ class SynchronizingRenamedProject extends ProjectSynchronizationSpecification {
         SynchronizationResult result = trySynchronizeAndWait(sample)
 
         then:
-        result.status.exception instanceof UnsupportedConfigurationException
+        result.status.exception instanceof BuildActionFailureException
```

Comment:
❌ So we can't deduplicate root projects, but why? In this case, why not rename the project to `already-there-2`? If that's a hard limitation that we can't fix for Gradle 5.5 then the thrown exception should be `UnsupportedConfigurationException`.

Reply:
`HierarchicalElementDeduplicator` in its current form doesn't support it:
https://github.com/gradle/gradle/blob/e975872ae2cf1d521b73f1107cae772ca09c63cd/subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/configurer/HierarchicalElementDeduplicator.java#L40-L43
I guess we can check the Exception if it contains the `IllegalArgumentException` and throw an `UnsupportedConfigurationException`